client/app/bundles/shared/components/Dropdown/DropdownFactory.jsx

```diff
@@ -0,0 +1,36 @@
+// @flow
+import React from 'react';
+import css from './Dropdown.scss';
+
+type Option = {
+  key?: string,
+  label: string,
+  value: any,
+};
+
+type Props = {
+  onChange?: (event: Event) => any,
+  options: Option[],
+  value?: any,
+};
+
+const DropdownFactory = (variationClassName: string) => {
+  const DropdownComponent = ({ onChange, options, value: propValue }: Props) => (
+    <select className={`${css.dropdown} ${variationClassName}`} onChange={onChange} value={propValue}>
+      {options.map(({ key, label, value }: Option) => (
+        <option key={key || (value !== null && value !== undefined ? value : label)} value={value}>
```

Comment:
Try replacing `(value !== null && value !== undefined ? value : label)` with `value || label`

Reply:
Actually I should just use id generator when key is not provided. It's possible for two options to have same value.